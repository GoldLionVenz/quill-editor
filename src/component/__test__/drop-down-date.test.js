import React from "react"
import ReactDOM from "react-dom";
import DropDownBreak from "../drop-down-date";
import SmmlEditor from "../index"
import { render, cleanup } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"

import renderer from "react-test-renderer"
afterEach(cleanup)

it("renders without crashing", ()=>{
    const div = document.createElement("div")
    ReactDOM.render(<DropDownBreak/>,div)
}) 

it("renders select correctly", ()=>{
    const { getByTestId } = render(<DropDownBreak/>)

    expect(getByTestId("select")).toHaveTextContent("mdy")
})

it("renders select correctly", ()=>{
    const { getByTestId } = render(<DropDownBreak/>)

    expect(getByTestId("select")).toHaveTextContent("ymd")
})

it("matches snapshot", ()=>{
    const tree = renderer.create(<DropDownBreak/>).toJSON();
    expect(tree).toMatchSnapshot()
})