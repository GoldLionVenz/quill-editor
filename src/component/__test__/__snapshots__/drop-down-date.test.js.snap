// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<span
  className="ql-ssml_date ql-picker "
  data-testid="select"
  onClick={[Function]}
  style={
    Object {
      "padding": "4px 0px 0px",
      "width": 45,
    }
  }
>
  <span
    className="ql-picker-label"
    role="button"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
      }
    }
    tabIndex="0"
  >
    <i
      className="calendar alternate outline icon"
      title="Date"
    />
    <svg
      viewBox="0 0 18 18"
    >
      <polygon
        className="ql-stroke"
        points="7 11 9 13 11 11 7 11"
      />
      <polygon
        className="ql-stroke"
        points="7 7 9 5 11 7 7 7"
      />
    </svg>
  </span>
  <span
    className="ql-picker-options"
    id="ql-picker-options-3"
    tabIndex="-1"
  >
    <button
      className="ql-insertDate ql-picker-item"
      value="mdy"
    >
      mdy
    </button>
    <button
      className="ql-insertDate ql-picker-item"
      value="dmy"
    >
      dmy
    </button>
    <button
      className="ql-insertDate ql-picker-item"
      value="ymd"
    >
      ymd
    </button>
    <button
      className="ql-insertDate ql-picker-item"
      value="md"
    >
      md
    </button>
    <button
      className="ql-insertDate ql-picker-item"
      value="dm"
    >
      dm
    </button>
    <button
      className="ql-insertDate ql-picker-item"
      value="ym"
    >
      ym
    </button>
    <button
      className="ql-insertDate ql-picker-item"
      value="my"
    >
      my
    </button>
    <button
      className="ql-insertDate ql-picker-item"
      value="d"
    >
      d
    </button>
    <button
      className="ql-insertDate ql-picker-item"
      value="m"
    >
      m
    </button>
    <button
      className="ql-insertDate ql-picker-item"
      value="y"
    >
      y
    </button>
  </span>
</span>
`;
